tests/probes_test.go

```diff
@@ -89,6 +89,64 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 			table.Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server", httpProbe, tests.StartHTTPServer),
 		)
 
+		table.DescribeTable("should succeed with 'Host' field", func(readinessProbe *v12.Probe) {
+
+			isHTTPProbe := readinessProbe.Handler.HTTPGet != nil
+			isDualStack := tests.IsIPv6Cluster(virtClient)
+
+			By("Create a pod with the readiness probe backend")
+			var probeBackendPod *v1.Pod
+			if isHTTPProbe {
+				probeBackendPod = tests.StartHTTPServerPod(1500)
+			} else {
+				probeBackendPod = tests.StartTCPServerPod(1500)
+			}
+			defer func() {
+				err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Delete(probeBackendPod.Name, &v13.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}()
+
+			By("Specifying a VMI with a readiness probe with 'Host' field")
+			vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			vmi.Spec.ReadinessProbe = readinessProbe
+			ipIndex := 0
+
+			// Select the ipv6 address if we are at dual stack
+			if isDualStack {
+				ipIndex = 1
+			}
+			selectedIP := probeBackendPod.Status.PodIPs[ipIndex].IP
```

Comment:
I think you should not make assumptions about the indexes of the IP families; now it's like this, in the future it will be configurable.

I know it's an extra step, but checking which is the ipv6 address will be a longer lasting solution.